### PR TITLE
Adding Typescript definition file

### LINKED ISSUE #53: Adding Typescript Type Declaration File
Hello! I'm trying to use your module as part of a typescript project, but unfortunately it is missing a Typescript type declaration. I added the module like this to my code (I'm using webpack):

`import { fetch } from "fetch-ponyfill";`

But I'm getting the following error when trying to compile Typescript to JS: 

`(9,23): error TS7016: Could not find a declaration file for module 'fetch-ponyfill'. '/Users/simon.kreiser/Development/frontend/node_modules/fetch-ponyfill/fetch-node.js' implicitly has an 'any' type.`

Would you be interested in adding Typescript support? I'd be willing to help out or contribute. I'm relatively new to typescript, but I found this tutorial on how to write d.ts. files: https://blogs.msdn.microsoft.com/typescript/2016/12/14/writing-dts-files-for-types/

### PR DESCRIPTION
**Pull request by duclet**
Resolves #53.

I've only recently started toying around with Typescript and felt the need for this. With minimal testing, this had worked for me but I haven't have that much experience with declaration files. Perhaps someone can test and verify it is works for them too or not.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
+declare module "fetch-ponyfill" {
+    export default FetchPonyfill.bootstrap;
```
Comment (duclet) [inline index.d.ts]: All the documentation I find says that this should be `export = FetchPonyfill.bootstrap` but yet that does not work for me, only this does.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    interface FetchPonyfillStatic {
+        fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
+        Headers: typeof Headers;
+        Request: typeof Request;
+        Response: typeof Response;
+    }
+
+    type HeadersInit = Headers | string[][] | { [key: string]: string };
```
Comment (duclet) [inline index.d.ts]: Pretty much everything starting from here is copied over from @types/whatwg-fetch

**Comment by duclet** [comment]
Updated the definition. Turns out the Fetch API is already part of Typescript's lib. So rather than re-defining it, I've updated it to just define the bootstrap function this module exposes and have it returns the objects from the lib.

**Comment by qubyte** [comment]
Very nice (and much simpler for me to parse with the update). I'm going to pore over it for a little while to make sure I understand. :)

**Comment by qubyte** [comment]
Nope, looks good. Just been a busy week.

**Pull request merged**